detectable/src/main/java/com/synopsys/integration/detectable/detectables/cpan/CpanCliExtractor.java

```diff
@@ -38,4 +40,18 @@ public Extraction extract(ExecutableTarget cpanExe, ExecutableTarget cpanmExe, F
         CodeLocation detectCodeLocation = new CodeLocation(dependencyGraph);
         return new Extraction.Builder().success(detectCodeLocation).build();
     }
+
+    List<String> generateCpanListOutput(File workingDirectory, ExecutableTarget cpanExe) throws ExecutableRunnerException {
+        Map<String, String> environmentVariables = new HashMap<>();
+
+        // When cpan is run on a system for the first time, the user is presented with a number of prompts; this can cause Detect to wait indefinitely.
+        // The following line causes the execution to accept defaults and does not block on any prompts so that Detect can complete its execution:
+        environmentVariables.put("PERL_MM_USE_DEFAULT", "true");
```

Comment:
does this work in an environment where the customer might not be connected to the internet? Not sure if that is supported by this particular detector but was wondering what might happen if the the user is offline and we try to take a default of yes for this question:

Would you like me to automatically choose some CPAN mirror
sites for you? (This means connecting to the Internet) [yes]

Reply:
Good question. I haven't thought about this before so I ran a few tests.

Before each test I disconnected from the Internet and I ran `rm -rf ~/cpan/*` which causes `cpan -l` to hang on a prompt.

### Testing `cpan -l` in isolation

#### Responding to prompts manually

The following was displayed after I accepted automatically choosing CPAN mirror:
```
Would you like me to automatically choose some CPAN mirror
sites for you? (This means connecting to the Internet) [yes] 
curl: (6) Could not resolve host: www.perl.org
curl: (6) Could not resolve host: ftp.perl.org
curl: (6) Could not resolve host: www.perl.org
curl: (6) Could not resolve host: ftp.perl.org
CPAN.pm needs at least one URL where it can fetch CPAN files from.

Please enter your CPAN site: [] 
CPAN.pm needs at least one URL where it can fetch CPAN files from.

Please enter your CPAN site: [] 
CPAN.pm needs at least one URL where it can fetch CPAN files from.

Please enter your CPAN site: [] 
CPAN.pm needs at least one URL where it can fetch CPAN files from.

Please enter your CPAN site: [] 
CPAN.pm needs at least one URL where it can fetch CPAN files from.

Please enter your CPAN site: []
```

After answering these questions, however, the command finished successfully with a zero exit code.

#### Accepting defaults

When accepting defaults the command ran successfully.

### Testing as part of Detect execution

I was able to scan a project successfully with this change while not connected to the Internet. Components were picked up and ended up in the BDIO.

### Conclusion

I believe this change will not cause any negative behaviour when not connected to the Internet and is an improvement overall.

Having said that, I don't think we can claim that this detector is guaranteed to work offline. Since it executes a build, the build could depend on fetching something from the Internet.

Reply:
agree this is definitely an improvement overall, going to approve on that basis